src/adapters/supabase/helpers/client.ts

```diff
@@ -206,28 +206,48 @@ export const upsertAccessControl = async (username: string, repository: string,
   const logger = getLogger();
   const { supabase } = getAdapters();
 
-  const { data, error } = await supabase.from("access").select("user_name").eq("user_name", username).single();
+  const { data, error } = await supabase.from("access").select("user_name").eq("user_name", username).eq("repository", repository).single();
 
   const properties = { user_name: username, repository: repository, updated_at: new Date().toUTCString(), [access]: bool };
 
   if (data) {
     const { data: _data, error: _error } = await supabase.from("access").upsert(properties);
     logger.info(`Upserting an access done, { data: ${data}, error: ${error} }`);
   } else {
-    const { data: _data, error: _error } = await supabase
-      .from("access")
-      .insert({
-        created_at: new Date().toUTCString(),
-        price_access: false,
-        time_access: false,
-        multiplier_access: false,
-        priority_access: false,
-        ...properties,
-      });
+    const { data: _data, error: _error } = await supabase.from("access").insert({
+      created_at: new Date().toUTCString(),
+      price_access: false,
+      time_access: false,
+      multiplier_access: false,
+      priority_access: false,
+      ...properties,
+    });
     logger.info(`Creating a new access record done, { data: ${_data}, error: ${_error} }`);
   }
 };
 
+/**
+ * Queries the access table
+ *
+ * @param username The username you want to find access for
```

Comment:
Okay, I'd remove it.. Was supposed to be for function description on hover

Reply:
Interesting use case. I suppose I've been relying on Copilot for too long where it will automatically fill in the correct values based on the function signature itself (No mouse over needed!)